mirascope/core/mistral/stream.py

```diff
@@ -38,3 +43,45 @@ def _construct_message_param(
             role="assistant", content=content if content else "", tool_calls=tool_calls
         )
         return message_param
+
+    def construct_call_response(self) -> MistralCallResponse:
+        if self.message_param is None:
+            raise ValueError(  # pragma: no cover
+                "No stream response, check if the stream has been consumed."
+            )
+        usage = UsageInfo(prompt_tokens=0, completion_tokens=0, total_tokens=0)
```

Comment:
why not just do something like:
```python
usage = UsageInfo(
    prompt_tokens=(self.input_tokens or 0),
    completion_tokens=(self.output_tokens),
    total_tokens=((self.input_tokens or 0) + (self.output_tokens or 0))
)
```
Maybe we can do this in the other providers as well

Reply:
i have to check for float, so i think what i'm doing is cleaner